test/model/src/include/model/driver/debug_log_parser.h

```diff
@@ -92,6 +92,23 @@ class debug_log_parser {
         inline row_remove() : fileid(0) {}
     };
 
+    /*
+     * debug_log_parser::row_truncate --
+     *     The row_truncate log entry.
+     */
+    struct row_truncate {
+        uint64_t fileid;
+        uint32_t mode;
+        std::string start;
+        std::string stop;
+
+        /*
+         * debug_log_parser::row_truncate::row_truncate --
+         *     Default constructor.
+         */
+        inline row_truncate() : fileid(0) {}
```

Comment:
General question about these types - there are some inconsistencies with what gets an initialiser and what doesn't. It looks like `fileid` needs to be zero by default, but the `prev_lsn` type breaks that rule. I also can't see other code that would indicate `fileid` is special/assumed to be valid - why not initialise `mode` too?

Reply:
Nice catch; Coverity would definitely complain about that.

I haven't been adding the constructors as they are functionally not necessary, because all fields will be filled in when the corresponding parts of the debug log are parsed. But Coverity complains that when a struct contains non-primitive types (e.g., `std::string`), the constructor that C++ would generate would initialize only the strings and not fields with primitive types, so hence the need for these constructors. This apparently isn't an issue for structs with just primitive types...

Reply:
Ah, that's the difference then - thanks for explaining.